src/openclsim/plot/superlog.py

```diff
@@ -0,0 +1,459 @@
+                ].tolist()
+                x_list = [[s, e, e] for s, e in zip(start_times, end_times)]
+                x = [item for sublist in x_list for item in sublist]
+                y = [source_object, source_object, None] * len(x)
+                traces.append(
+                    go.Scatter(
+                        name=activity,
+                        x=x,
+                        y=y,
+                        mode="lines",
+                        hoverinfo="y+name",
+                        line=dict(color=color, width=10),
+                        connectgaps=False,
+                    )
+                )
+
+        # add critical path based on 'names'
+        if "is_critical" in list(self.df_super_log.columns):
+            x_critical = self.df_super_log.loc[
+                self.df_super_log.loc[:, "is_critical"], "start_time"
+            ].tolist()
+            y = self.df_super_log.loc[
+                self.df_super_log.loc[:, "is_critical"], "SourceObject"
+            ].tolist()
+            traces.append(
+                go.Scatter(
+                    name="critical_path",
+                    x=x_critical,
+                    y=y,
+                    mode="markers",
+                    hoverinfo="name",
+                    line=dict(color="Black", width=5),
+                    connectgaps=False,
+                )
+            )
+
+        layout = go.Layout(
+            title="GANTT Chart",
+            hovermode="closest",
+            legend=dict(x=0, y=-0.2, orientation="h"),
+            xaxis=dict(
+                title="Time",
+                titlefont=dict(
+                    family="Courier New, monospace", size=18, color="#7f7f7f"
+                ),
+                range=[
+                    min(self.df_super_log.loc[:, "start_time"]),
+                    max(self.df_super_log.loc[:, "end_time"]),
+                ],
+            ),
+            yaxis=dict(
+                title="Activities",
+                titlefont=dict(
+                    family="Courier New, monospace", size=18, color="#7f7f7f"
+                ),
+            ),
+        )
+
+        if static is False:
+            init_notebook_mode(connected=True)
+            fig = go.Figure(data=traces, layout=layout)
+
+            return iplot(fig, filename="news-source")
+        else:
+            return {"data": traces, "layout": layout}
+
+    @classmethod
+    def from_objects(cls, objects, id_map=None):
+        """get logs form objects and init object"""
+        # extract logs from all objects
+        combined_log = combine_logs(objects, id_map=id_map)
+        # reshape into set of activities with start, duration and end
+        super_log = reshape_superlog(combined_log)
+        # add unique identifier for activities (may be shared by multiple objects)
+        super_log = add_unique_activity(super_log)
+        return cls(df_super_log=super_log)
+
+
+def reshape_superlog(super_log):
```

Comment:
Why did you not make these (reshape_superlog, add_unique_activity, combine_logs, get_superlog_with_critical_path) methods of the class?


I think ```reshape_superlog```, ```add_unique_activity```, and ```combine_logs``` can be classmethods just like ```from_objects```

and ```get_superlog_with_critical_path``` can be just a method